### PR TITLE
Remove Status.value

### LINKED ISSUE #1744: The spec doesn't seem to define numeric values for the status codes, do we need these?
https://www.github.com/open-telemetry/opentelemetry-specification/tree/master/specification%2Ftrace%2Fapi.md

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #1744 

Not quite merged but https://github.com/open-telemetry/opentelemetry-specification/pull/1257 has so many approvals I just went with its spec.

### CODE REVIEW

**Comment by Oberon00** [comment]
This PR also changes attribute names in the Jaeger exporter. Intentionally?

**Comment by anuraaga** [comment]
Yeah went ahead and reflected the almost merged https://github.com/open-telemetry/opentelemetry-specification/pull/1257 since it makes clearest what to do instead of using value.

**Review by Oberon00 — APPROVED** [review]
I think this makes sense. 👍

**Review by jkwatson — APPROVED** [review]
Thanks!

File: api/src/main/java/io/opentelemetry/api/trace/StatusCode.java [hunk 1]
```diff
@@ -5,36 +5,18 @@
...
-  StatusCode(int value) {
-    this.value = value;
-  }
-
-  /**
-   * Returns the numerical value of the code.
-   *
-   * @return the numerical value of the code.
-   */
-  public int value() {
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/api/trace/StatusCode.java:37]: since this is a public API should we deprecate it before deleting?

Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/api/trace/StatusCode.java:37]: Makes sense

**Pull request merged**